From: Michael Ellerman <michael@ellerman.id.au>
Subject: [PATCH] powerpc: Initialise hvlpevent_queue.lock correctly

When I changed the hvlpevent_queue code to use a spinlock instead of a custom
atomic (719d1cd86780c156f954fc34f34481adac197aec) I didn't initialise the
lock anywhere, oops.

Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
Signed-off-by: Olaf Hering <olh@suse.de>
---

 arch/powerpc/platforms/iseries/lpevents.c |    2 ++
 1 files changed, 2 insertions(+)

Index: iseries/arch/powerpc/platforms/iseries/lpevents.c
===================================================================
--- iseries.orig/arch/powerpc/platforms/iseries/lpevents.c
+++ iseries/arch/powerpc/platforms/iseries/lpevents.c
@@ -184,6 +184,8 @@ void setup_hvlpevent_queue(void)
 {
 	void *eventStack;
 
+	spin_lock_init(&hvlpevent_queue.lock);
+
 	/* Allocate a page for the Event Stack. */
 	eventStack = alloc_bootmem_pages(LpEventStackSize);
 	memset(eventStack, 0, LpEventStackSize);
